samples/openthread/coap_server/src/ot_coap_utils.c

```diff
@@ -0,0 +1,249 @@
+#include <logging/log.h>
+#include <net/net_pkt.h>
+#include <net/net_l2.h>
+#include <net/openthread.h>
+#include <openthread/coap.h>
+#include <openthread/ip6.h>
+#include <openthread/message.h>
+#include <openthread/thread.h>
+
+#include "ot_coap_utils.h"
+
+LOG_MODULE_REGISTER(ot_coap_utils, CONFIG_OT_COAP_UTILS_LOG_LEVEL);
+
+struct server_context {
+	struct otInstance *ot;
+	bool provisioning_enabled;
+	light_request_callback_t on_light_request;
+	provisioning_request_callback_t on_provisioning_request;
+};
+
+static struct server_context srv_context = { .ot = NULL,
+					     .provisioning_enabled = false,
+					     .on_light_request = NULL,
+					     .on_provisioning_request = NULL };
+
+/**@brief Definition of CoAP resources for provisioning. */
+static otCoapResource provisioning_resource = { .mUriPath =
```

Comment:
A nit really, but don't you think that the following:
```
static otCoapResource provisioning_resource = {
	.mUriPath = PROVISIONING_URI_PATH,
	.mHandler = NULL,
	.mContext = NULL,
	.mNext = NULL,
};
```
looks better? If not, nevermind :)

Reply:
It is not really my decision. To avoid this kind of discussions I use clang-formatter with Zephyr `.clang-format` configuration file.

Reply:
I understand `clang-format` does this, but it's also a known issue this is not correct. I've investigated it some time ago, you can prevent clang-format from messing up the struct initialization by adding a trailing comma to the last entry (as in my example). At least it used to help some time ago.

Reply:
Ok, it helps. Changed.